### DISCUSSION TITLE
New install setup questions

### DISCUSSION THREAD

**Discussion by slvrsky07** [basking-in-the-sun2000/solar-logger#59] [General]
Hello, first of all I think your project is great. I have been looking for something open source and customizable like you made here. I currently have a Megarevo inverter connected to a RPi and been controlling the settings using minimalmodbus through USB RS485. I have the list of factory registers (I can share them with you if you are interested) and shouldn't have an issue changing the inverter definitions file you have to match my inverters registers to your list.

Is it possible to change the pymodbus to use the usb serial connection instead of the TCP internet based polling? I assume this is why I am getting an [Errno 98] when trying to test the program.

Thank you for your time and making this product!

**Comment by basking-in-the-sun2000** [comment]
:)

The biggest challenge you have is that the modbus is rather dumb. It only allows for only one master on the bus, so you can't have the rpi sending commands along the inverter. Pymodbus allows for rtu connections, so it should be possible. 

One of the latest releases added the option to use rtu (inverter_type = "RTU" in config.py, also see modbus.py). Not sure if that would fix your problems, since I don't have a way to test it :(

You might also have to set the other options in the config file
`connection_port = "/dev/ptyp0"
connection_baudrate = 9600
connection_stopbits = 1
connection_bytesize = 8
connection_parity = 'E'`

**Reply by slvrsky07** [reply]
Thanks I’ll try that. Also do you have a list of the required registers you need to have this work? My inverter doesn’t have a lot of the same ones. I’m having an issue with the derating_w error.

**Comment by basking-in-the-sun2000** [comment]
take a look at https://github.com/basking-in-the-sun2000/solar-logger/issues/47

You can use the huawei_default as a guide
